client/test/Completions.test.ts

```diff
@@ -38,6 +38,25 @@ describe('Completions', () => {
         await pollUntil(() => vscode.window.visibleTextEditors.length === 0, 1000);
     });
 
+    it('Can get HTML completions on document open', async () => {
+        // This test relies on the Index.cshtml document containing at least 1 HTML tag in it.
+        // For the purposes of this test it locates that tag and tries to get the Html completion
+        // list from it.
+
+        const content = doc.getText();
+        const tagNameIndex = content.indexOf('<') + 1;
+        const docPosition = doc.positionAt(tagNameIndex);
+        const completions = await vscode.commands.executeCommand<vscode.CompletionList>(
+            'vscode.executeCompletionItemProvider',
```

Comment:
Since this is a functional test(from what you said in the description), wouldn't it be better to send a keystroke (ctrl+j) at this location instead of invoking the CompletionProvider ourselves?

Reply:
Wish we could but sadly can't T_T